sdk/go/api_esc_extensions.go

```diff
@@ -29,13 +34,90 @@ func NewAuthContext(accessToken string) context.Context {
 	)
 }
 
+// DefaultAuthContext creates a new context, retrieving Pulumi Access Token
+// from either PULUMI_ACCESS_TOKEN environment variable, or from
+// currently logged in account in Pulumi CLI or ESC CLI
+//
+// This context can be used to authenticate requests to the ESC API.
+func NewDefaultAuthContext() (context.Context, error) {
+	accessToken := os.Getenv("PULUMI_ACCESS_TOKEN")
+	if accessToken != "" {
+		return NewAuthContext(accessToken), nil
+	}
+
+	workspace := esc_workspace.New(esc_workspace.DefaultFS(), esc_workspace.DefaultPulumiWorkspace())
+	account, _, err := workspace.GetCurrentAccount(false)
+	if err != nil {
+		return nil, fmt.Errorf("Error grabbing current account: %w", err)
+	}
+	if account != nil {
+		return NewAuthContext(account.AccessToken), nil
+	}
+
+	return nil, errors.New("No default Pulumi Access Token found. Either export PULUMI_ACCESS_TOKEN " +
+		"environment variable, or login using Pulumi or ESC CLI.")
+}
+
 // NewClient creates a new ESC client with the given configuration.
 func NewClient(cfg *Configuration) *EscClient {
 	client := &EscClient{rawClient: NewRawAPIClient(cfg)}
 	client.EscAPI = client.rawClient.EscAPI
 	return client
 }
 
+// NewCustomBackendConfiguration creates a new Configuration object,
+// but replaces default API endpoint with a given custom backend URL
+func NewCustomBackendConfiguration(customBackendURL url.URL) *Configuration {
+	if !strings.HasSuffix(customBackendURL.String(), "/api/esc") {
+		appendedURL := customBackendURL.JoinPath("api", "esc")
+		customBackendURL = *appendedURL
+	}
```

Comment:
nit but I think this would be safer than do check for the suffix
```suggestion
	appendedUrl, err := url.Parse(fmt.Sprintf("%s://%s/api/esc", customBackendURL.Scheme, customBackendURL.Hostname()))
	if err != nil {
	    //....
```

Reply:
This is nice - ensures there's nothing extra - no more no less. Will do!